app/components/file_attachment_list/file_attachment_list.js

```diff
@@ -71,17 +70,14 @@ export default class FileAttachmentList extends PureComponent {
         }
     }
 
-    componentWillReceiveProps(nextProps) {
-        if (this.props.files !== nextProps.files) {
-            this.filesForGallery = this.getFilesForGallery(nextProps);
+    componentDidUpdate(prevProps) {
+        if (prevProps.files !== this.props.files) {
+            this.filesForGallery = this.getFilesForGallery(this.props);
             this.buildGalleryFiles().then((results) => {
                 this.galleryFiles = results;
             });
         }
-        if (!this.state.loadingFiles && nextProps.files.length === 0) {
```

Comment:
`loadingFiles` property is not using anywhere except in componentWillReceiveProps and there I think it's redundant, so, I removed it. Is it wrong?

Reply:
You're correct that it doesn't need to be in the state, but we still need to track if there was a change in the props so that we don't call `loadFileForPost` repeatedly any time the component updates while `this.props.files.length` is still 0. You can do that by comparing against the previous value of the `files` prop though by changing this to
```javascript
if (this.props.files !== prevProps.files && this.props.files.length === 0) {}
```